src/properties.ts

```diff
@@ -0,0 +1,219 @@
+   *   defaultValue: new Date(),
+   *  },
+   */
+  DATE: 'date',
+
+  /**
+   * color: {
+   *   type: 'color',
+   *   defaultValue: '#000000',
+   * },
+   */
+  COLOR: 'color',
+
+  /**
+   * button: {
+   *  type: 'button',
+   *   onClick: () => {
+   *    ... code to modify some variables
+   *  }
+   * },
+   */
+  BUTTON: 'button',
+
+  /**
+   * otherStyles: {
+   *   type: 'object',
+   *   label: 'Styles',
+   *   defaultValue: {
+   *     border: '2px dashed silver',
+   *     borderRadius: 10,
+   *     padding: 10,
+   *   },
+   * },
+   */
+  OBJECT: 'object',
+
+  /**
+   * items: {
+   *   type: 'array',
+   *   label: 'Items',
+   *   defaultValue: ['Laptop', 'Book', 'Whiskey'],
+   * },
+   */
+  ARRAY: 'array',
+
+  /**
+   * images: {
+   *   type: 'files',
+   *   label: 'Happy Picture',
+   *   accept: 'image/*',
+   *   defaultValue: [
+   *     'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAQAAAC1+jfqAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QA/4ePzL8AAAAHdElNRQfiARwMCyEWcOFPAAAAP0lEQVQoz8WQMQoAIAwDL/7/z3GwghSp4KDZyiUpBMCYUgd8rehtH16/l3XewgU2KAzapjXBbNFaPS6lDMlKB6OiDv3iAH1OAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE4LTAxLTI4VDEyOjExOjMzLTA3OjAwlAHQBgAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOC0wMS0yOFQxMjoxMTozMy0wNzowMOVcaLoAAAAASUVORK5CYII=',
+   *   ],
+   * },
+   */
+  FILES: 'files',
+};
+
+export interface StoryPropertyObject {
+  type?: string;
+
+  /**
+   * label to display next to the field editor
+   * by default uses the property name itself
+   */
+
+  label?: string;
+
+  /**
+   * placehlder for empty properties
+   * either undefined defautValue
+   * or user clears the field
+   */
+  placeholder?: string;
+
+  /**
+   * a default value for the property
+   */
+  defaultValue?: any;
+
+  /**
+   * hide the label from the property editor
+   */
+  hideLabel?: boolean;
+  /**
+   * hide the property editor for this property
+   * will only use the defaultValue
+   */
+
+  hidden?: boolean;
+  /**
+   * allows grouping of the properties
+   * in a property editor
+   * for example different editor tabs
+   */
+  groupId?: string;
+
+  // FIELD TYPE SPECIFIC
+  /**
+   * for button type fields, an onClick handler
+   */
+  onClick?: () => any;
+
+  /**
+   * for options type fields
+   */
+
+  display?: 'select' | 'radio' | 'inline-radio' | 'multi-select' | 'check' | 'inline-check';
+
+  /**
+   * for numeric type fields
+   */
+
+  /**
+   * if true, will display a range type slider editor
+   */
+  range?: boolean;
+
+  /**
+   * minimum allowed value for numeric propery
+   */
+  min?: number;
+
+  /**
+   * maximum allowed value for numeric propery
+   */
+  max?: number;
+
+  /**
+   * stepper for numeric editor /i nc/dec value
+   */
+
+  step?: number;
+}
+
+/**
+ * StoryProperty is a either an object of property settings
+ * or a shortcut can be used:
+ *  properties: {
+ *   text: 'Hello',
+ * },
+ */
+
+export type StoryProperty = StoryPropertyObject | string;
```

Comment:
when/why would StoryProperty === string?

Reply:
StoryPropertyObject | string; is for a property definition shortcut
example : 
```
name: "Tom"
```
would be a 'text' type field with a defaultValue of "Tom"

Reply:
Are we going to have this shortcut only for strings?

I get it's kinda nice, but it will also open the floodgates for request to support the shortcut for `number` `Date`, etc.
Do we want to support them all?

I wouldn't mind keeping this a bit strict.

Reply:
remember CSF isn't just for storybook, ALL places using CSF would have to support the shorthands

Reply:
I am open either way

Reply:
@shilman WDYT about keeping this strict verses having a few shorthands?

Reply:
Btw, the shorthand feature was requested by @tmeasday on the google docs brainstorming. @jonspalmer mentioned deducting field type from field value to avoid having to enter it.

Reply:
I am myself leaning a bit more towards strict as suggested by @ndelangen , but fine either way

Reply:
I'm a big fan of the shorthand. IMO its going to be 80%+ of the use cases. We'll need 'any' to support the shorthand.

It would be relatively straight forward for us to provide a util method in this repo that converts the shorthand to the longhand. It feels like a great tradeoff for consumers of CSF to have to call that util for the benefit of the more ergonomic shorthand.